### ISSUE TITLE
Leaking gtags processes?

### ISSUE THREAD

**Issue by ljusten** [austin-----/code-gnu-global#22]
I'm using VS Code and this plugin on the (huge) Chromium code base and I'm noticing that over time more and more instances of gtags processes accumulate in my system monitor (Ubuntu 14.04). Over time, they eventually suck up 100% of my CPU resources and make my system sluggish to use. gtags runs for a long time, so I'm wondering whether the plugin just keeps kicking gtags regularly at a rate faster than instances of gtags finish, so they accumulate.

**Comment by austin-----** [comment]
@ljusten This plugin does not execute "gtags" which is the tagging program. It only executes "global" which queries the GTAGS db file.

**Comment by tsafin** [comment]
Yes, this is exact observation over my huge repository - it launches multiple global processes, which use gtags processes, and eventually they all eat 100% of time. 
The question is - why they have never completed?

**Comment by austin-----** [comment]
It looks like there was a merge that introduced auto executing gtags on save. We should probably make this optional so that big project can opt out of this. I guess some version of gtags have issues finishing big projects in time.

**Comment by csholmq** [comment]
Which is weird, since the gtags call triggered by filesave incremental update. Should be fast.

Perhaps this can be solved with another gtags version or maybe a semaphore, preventing more processes to spawn until the first has finished.
